docs/api/IoT.js-API-UART.md

```diff
@@ -1,46 +1,90 @@
-## Module: uart
-
 ### Platform Support
 
 The following shows uart module APIs available for each platform.
 
 |  | Linux<br/>(Ubuntu) | Raspbian<br/>(Raspberry Pi) | Nuttx<br/>(STM32F4-Discovery) |
 | :---: | :---: | :---: | :---: |
 | uart.open | O | O | O |
-| uart.close | O | O | O |
-| uart.write | O | O | O |
+| uartport.write | O | O | O |
+| uartport.writeSync | O | O | O |
+| uartport.close | O | O | O |
+| uartport.closeSync | O | O | O |
 
-### Constructor
 
-#### new uart(path[, options][, callback]) for Ubuntu, Raspbian
-#### new uart(path[, callback]) for Nuttx
-* `path: String`
-* `options: Object`
-* `callback: Function(err)`
+## Contents
+  * [UART](#uart)
+    * [Constructor](#constructor)
+      * [`new UART()`](#new-uart)
+    * [Prototype Methods](#uart-prototype-methods)
+      * [`uart.open(configurable[, callback])`](#uart-open)
+  * [UARTPort](#uartport)
+    * [Prototype Methods](#uartport-prototype-methods)
+      * [`uartport.write(data[, callback])`](#uartport-read)
+      * [`uartport.writeSync(data)`](#uartport-read-sync)
+      * [`uartport.close([callback])`](#uartport-close)
+      * [`uartport.closeSync()`](#uartport-close-sync)
+    * [Event](#uartport-event)
 
-`options` is an object specifying following information:
-* `baudRate: Number (Default value is 9600)`
-* `dataBits: Number (Default value is 8)`
-* `autoOpen: Boolean (Default value is true)`
 
-* On Nuttx, you can set the properties of `options` only on Nuttx config. `Device Drivers -> Serial Driver Support -> U[S]ART(N) Configuration` will help. 
+## Class: UART <a name="uart"></a>
 
-Create UART object. Following methods can be called with UART object.
 
+## Constructor <a name="uart-constructor"></a>
 
-For example,
+
+### `new UART()` <a name="uart-new"></a>
+
+Returns a new UART object which can open UART port.
+
+
+## Prototype methods <a name="uart-prototype-methods"></a>
+
+
+### `uart.open(configuration[, callback])` <a name="uart-open"></a>
+* `configuration <Object>`
+  * `device <String>`, mandatory configuration
+  * `baudRate: Number (Default value is 9600)`
+  * `dataBits: Number (Default value is 8)`
+* `callback <Function(err: Error | null)>`
+* Returns: `<UARTPort>`
+
+On Nuttx, you can set the properties of `configuration` only on Nuttx config. `Device Drivers -> Serial Driver Support -> U[S]ART(N) Configuration` will help. And if you want to get information about `device`, please see below list.
+  * [stm32f4](../../targets/nuttx-stm32f4/Stm32f4dis.md#uart).
+
+Opens UART port with the specified configuration. Following methods can be called with UARTPort object.
+
+
+**Example**
 ```javascript
-var uart = require('uart');
-var options = {
+var Uart = require('uart');
+
+var uart = new Uart();
+var configuration = {
```

Comment:
Are these all the fields that can be set?
I'd like to know what options are there, and what options are supported by this particular port (i.e. speed range, CTS/RTS, XON/XOFF, ...).